File: console/frontend/src/types/graph.ts
export interface NodeDefinition {
  id: string;
  label: string;
  node_type: 'orchestrator' | 'agent' | 'sub_agent' | 'error_handler' | 'terminal';
  model_id?: string;
  model_provider?: string;
  position?: { x: number; y: number };
}

export interface EdgeDefinition {
  id: string;
  source: string;
  target: string;
  conditional: boolean;
  label?: string;
}

export interface GraphDefinition {
  nodes: NodeDefinition[];
  edges: EdgeDefinition[];
  subgraphs: Record<string, GraphDefinition>;
}

export interface HealthCheck {
  status: string;
  service: string;
  checks: Record<string, boolean>;
  forge_console?: string;
  error?: string;
}

export interface SchedulerJob {
  id: string;
  next_run: string;
  trigger: string;
}

File: console/backend/app/models/graph.py
"""Graph definition models for React Flow rendering."""

from __future__ import annotations

from pydantic import BaseModel


class NodeDefinition(BaseModel):
    id: str
    label: str
    node_type: str  # "orchestrator", "agent", "sub_agent", "error_handler", "terminal"
    model_id: str | None = None
    model_provider: str | None = None
    position: dict[str, float] | None = None


class EdgeDefinition(BaseModel):
    id: str
    source: str
    target: str
    conditional: bool = False
    label: str | None = None


class GraphDefinition(BaseModel):
    nodes: list[NodeDefinition]
    edges: list[EdgeDefinition]
    subgraphs: dict[str, "GraphDefinition"] = {}
